doc/source/development/internals.rst

```diff
@@ -31,32 +31,6 @@ There are functions that make the creation of a regular index easy:
 * :func:`period_range`: fixed frequency date range generated from a time rule or
   DateOffset. An ndarray of :class:`Period` objects, representing timespans
 
-The motivation for having an ``Index`` class in the first place was to enable
```

Comment:
Would be good to add a `.. warning::` block here noting that subclassing an Index is not officially supported and users should use the extension array interface instead